### ISSUE TITLE
max_request didn't work

### ISSUE THREAD

**Issue by ozgurhangisi** [swoole/swoole-src#2698]
Please answer these questions before submitting your issue. Thanks!

1. What did you do? If possible, provide a simple script for reproducing the error.

I set max_request to 1000000 but worker didn't restart. if I manually call $swoole_server->reload(); it works and worker_request_count start from zero. Did I miss something ? 

```php
$swoole_server->set([
    'dispatch_mode'=>1,
    'max_request'=>1000000,
    'daemonize'=>2,
    'log_file'=>'/var/log/swoole',
    'worker_num'=>2,
    'task_worker_num'=>2,
    'http_compression'=>true,
    'http_gzip_level'=>1,
    'open_tcp_nodelay'=>true
]);
```

Here is the stats. Worker Request count is more than 1 million and it's increasing. : 

```

STATS=Array
(
    [start_time] => 1563326717
    [connection_num] => 29
    [accept_count] => 20535
    [close_count] => 20506
    [worker_num] => 4
    [idle_worker_num] => 2
    [tasking_num] => 0
    [request_count] => 2221077
    [worker_request_count] => 1110540
    [worker_dispatch_count] => 1110541
    [coroutine_num] => 1
)

```

2. What did you expect to see?

worker_request_count should start from zero.

3. What did you see instead?

worker_request_count is increasing.

4. What version of Swoole are you using (show your `php --ri swoole`)?

Swoole => enabled
Author => Swoole Team <team@swoole.com>
Version => 4.4.1
Built => Jul 16 2019 08:55:59
coroutine => enabled
trace_log => enabled
epoll => enabled
eventfd => enabled
signalfd => enabled
cpu_affinity => enabled
spinlock => enabled
rwlock => enabled
sockets => enabled
openssl => OpenSSL 1.0.2k-fips  26 Jan 2017
http2 => enabled
pcre => enabled
zlib => enabled
mutex_timedlock => enabled
pthread_barrier => enabled
futex => enabled
mysqlnd => enabled
async_redis => enabled

Directive => Local Value => Master Value
swoole.enable_coroutine => On => On
swoole.enable_library => On => On
swoole.enable_preemptive_scheduler => Off => Off
swoole.display_errors => On => On
swoole.use_shortname => On => On
swoole.unixsock_buffer_size => 8388608 => 8388608

5. What is your machine environment used (including version of kernel & php & gcc) ?

PHP 7.3.7 (cli) (built: Jul  3 2019 11:30:22) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.3.7, Copyright (c) 1998-2018 Zend Technologies
    with Zend OPcache v7.3.7, Copyright (c) 1999-2018, by Zend Technologies

**Comment by ozgurhangisi** [comment]
By the way I noticed that swoole reloads workers after ~1.5 M requests for the case above but I don't know how swoole decide when to reload workers.

**Comment by sshymko** [comment]
Setting `max_request` is for each worker process, i.e. `worker_num=30 max_request=10` would reload all workers after 300 requests (as long as requests are distributed evenly, i.e. `dispatch_mode=1`).

**Comment by ozgurhangisi** [comment]
Hi @sshymko ,

I have 2 workers and Request count is 2221077. Each worker has more than a million requests.

**Comment by sshymko** [comment]
@ozgurhangisi 
Yeah, that sounds like a bug. It should not exceed 2000000. How do you get statistics?

**Comment by ozgurhangisi** [comment]
@sshymko 

$swoole_server->stats() gets statistics about usage.

**Comment by sshymko** [comment]
@ozgurhangisi 
Thanks, good to know about the feature. Maybe the stats are inaccurate? You may want to try to count requests programmatically to confirm it really exceeds 1000000 per worker process.

**Comment by ozgurhangisi** [comment]
@sshymko  It's accurate with my Load Balancer request count.

**Comment by doubaokun** [comment]
@ozgurhangisi @sshymko

This is a feature by design, the `max_request` is only can be used to restart the workers after finishing roughly `max_request` requests, but can not be used to calculate the accurate request number on each worker. 

The `max_request` on each worker is adjusted by adding a random number, the random number is between 0 and half of the `max_request` set at the server.

Doing so to avoid the case all the workers are reloading during a very short period which can cause the pressure to DB and external cache.

So you should see the described effect: `worker_request_count ` is between `max_request` and 1.5 x `max_request`.

**Comment by sshymko** [comment]
@doubaokun 
That makes sense. Thanks for the explanation!

Giving more control over the feature would allow more flexibility and help to avoid the confusion at the same time.

Sounds like the following logic is implemented internally:
```
worker_max_request = max_request + rand(0, max_request / 2)
```
Would be great to be able to configure the randomness via a setting, for example:
```
max_request_margin = max_request_margin ?: max_request / 2
worker_max_request = max_request + rand(0, max_request_margin)
```
Configuration `max_request_margin = 0` would disable the variation.

**Comment by doubaokun** [comment]
@sshymko 

The change may make it more complex.

Like to understand the use case of disabling the `worker_max_request variation` or setting the `max_request_grace`.

Could you please explain more?

**Comment by ozgurhangisi** [comment]
@doubaokun In my case I have a small memory leak on one of the extension that I use. Our system is so big. (hundreds of methods, many extensions, tens of classes, etc..) If you have a huge system like us it's not easy to find root cause and fix it. But our system should run without any problem. I use max_request as quick and dirty fix until I find the root cause.

When I noticed this memory leak I calculated how much memory leak that I have for 10000 requests approximately and I noticed that I should restart the worker after 1 Million requests. 

So I already know when I should restart my server, I do lots of calculations, I review memory usage etc but swoole adds extra number to my calculation. I don't think it's necessary. So I know that I should restart my workers after 1 million request but I should set max_request to around 700,000 to restart it at around 1 million request. So do we really need this extra calculation ?
